### PR TITLE
PLT-3266 Added channel header to create channel modal

### PR DESCRIPTION
**Pull request by DavidLu1997**
#### Summary
Added a section for the channel header to the create channel modal

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3266

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [ ] Added API documentation (required for all new APIs)
- [ ] Has driver changes that have been merged and package.json updated
- [ ] Has enterprise changes (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization files updated
- [ ] Touches critical sections of the codebase (auth, upgrade, etc.)

image

### CODE REVIEW

File: webapp/components/new_channel_modal.jsx [hunk 1]
```diff
@@ -90,6 +90,7 @@ class NewChannelModal extends React.Component {
     handleChange() {
         const newData = {
             displayName: ReactDOM.findDOMNode(this.refs.display_name).value,
+            header: ReactDOM.findDOMNode(this.refs.channel_header).value,
```
Comment (enahum) [inline webapp/components/new_channel_modal.jsx]: @DavidLu1997 by getting the reference with `this.refs.channel_header` you don't need ReactDOM, so feel free to remove it.

Also feel free to do the same thing with the other ones too.

**Comment by DavidLu1997** [comment]
@enahum removed `React.findDOMNNode`

**Comment by enahum** [comment]
+1

**Comment by coreyhulen** [comment]
+1

**Comment by jasonblais** [comment]
Tested the following (Windows 10, Chrome):
 - create public channel from "+" sign next to **Channels**
 - create public channel from **More...** at the bottom of public channels list
 - create private group from "+" sign next to **Private Groups**
 - updating headers after creating the channel

The only note I have is to add more spacing between the channel purpose and header. The distance between the `channel purpose help text` and `channel header input box` should match the distance between the `channel name help text` and `channel purpose input box`

image

**Comment by DavidLu1997** [comment]
@jasonblais I think that is more of a ticket for @asaadmahmoodspin

**Comment by jasonblais** [comment]
Thanks, have asked @asaadmahmoodspin to take a look

**Comment by asaadmahmoodspin** [comment]
@DavidLu1997 The issue with the spacing is as follows.
screen shot 2016-07-14 at 6 38 07 pm

The markup looks like the screenshot above.
Where as it should be:

```
<div class="form-group">..</div>
<div class="form-group">
    <div class="col-sm-3">..</div>
    <div class="col-sm-9">..</div>
</div>
<div class="form-group less">
    <div class="col-sm-3">..</div>
    <div class="col-sm-9">..</div>
</div>
```

**Comment by DavidLu1997** [comment]
@jasonblais @asaadmahmoodspin changes made, now looks like this:
image

**Comment by asaadmahmoodspin** [comment]
Looks good to me in that screenshot.

**Comment by jasonblais** [comment]
+1 for screenshots, and the tests I mentioned above

**Pull request merged**